src/android/mod.rs

```diff
@@ -25,16 +25,19 @@ pub mod android {
     extern crate jni;
 
     use super::*;
+    use std::path::{PathBuf};
+    use rocket::config::{Config, Environment};
     use self::jni::JNIEnv;
     use self::jni::objects::{JClass, JString};
     use self::jni::sys::{jstring, jdouble};
+
     use datastore::Datastore;
     use models::{Event, Bucket};
+    use endpoints;
 
-    fn openDatastore() -> Datastore {
-        let db_dir = dirs::db_path().to_str().unwrap().to_string();
-        Datastore::new(db_dir)
-    }
+
+    static mut DATASTORE: Option<&Datastore> = None;
```

Comment:
I've seen this way of making Singletons with Options a few times before in a couple of gtk-rs projects, not a perfect solution but it's fine.

Reply:
Just realized though that those projects have them inside a atomic reference counter and a mutex which is not possible here.

Reply:
The best solution would probably be so DatabaseWorker is a singleton as the actual Database object is just a wrapper with a Requester from mpsc_requests which could be cloned without any issues.